### ISSUE TITLE
[Question] Any thoughts about adding vanilla algorithms?

### ISSUE THREAD

**Issue by skandermoalla** [DLR-RM/stable-baselines3#1384]
### ❓ Question

Hello,

I was wondering why vanilla methods such as REINFORCE (vanilla policy gradient without bootstrap i.e. Monte Carlo estimates), REINFORCE with a baseline (with a critic to estimate the baseline), or vanilla actor-critic (critic used to bootstrap) were not implemented in the library.
I guess some of these methods can be obtained by tweaking hyperparameters in already implemented algorithms, but it's not immediately clear how to do so.
I think proper implementations of those methods or documentation on how to get to them could be highly valuable to people making experiments isolating the effects of SoTA methods.
Has this been discussed before?

Thanks!

### Checklist

- [X] I have checked that there is no similar [issue](https://github.com/DLR-RM/stable-baselines3/issues) in the repo
- [X] I have read the [documentation](https://stable-baselines3.readthedocs.io/en/master/)
- [X] If code there is, it is minimal and working
- [X] If code there is, it is formatted using the [markdown code blocks](https://help.github.com/en/articles/creating-and-highlighting-code-blocks) for both code and stack traces.

**Comment by araffin** [comment]
Hello,

A2C code is actually quite close to vanilla actor critic (the policy loss is slightly different, but the critic update uses MC estimates).
I think one reason it has not been discussed before is because SB3 is targeted towards applications. If it was targeted towards education, it would properly include them as spinning up does: https://github.com/openai/spinningup

Another reason is that A2C already performs poorly on many RL benchmarks, so having an even simpler algorithm that probably performs worse is of little interest to users.

That being said, if you want an implementation of REINFORCE with SB3, you can find one here: https://github.com/osigaud/stable-baselines3/tree/master/stable_baselines3/reinforce
We created it for a RL course.

**Comment by araffin** [comment]
> but instead need to isolate components in SoTA vs vanilla algorithms and plug some additional logging and evaluation metrics?

For slight modifications, adding logging/metrics, SB3 should be fine (see SB3 contrib, where we have PPO variants for instance).
The code should be also simple enough to activate/deactivate features, however SB3 is not meant to be modular.

>  What other libraries do you think could be helpful in this case?

You can search for libraries that advertise themselves as modular, like https://github.com/pytorch/rl or https://github.com/takuseno/d3rlpy

You could also have a look at single file implementation from https://github.com/vwxyzjn/cleanrl

**Comment by skandermoalla** [comment]
Thanks!
